### DISCUSSION TITLE
Would like to help out writing docs on the internals.

### DISCUSSION THREAD

**Discussion by emilebosch** [ElMassimo/vite_ruby#94] [Q&A] [answered]
Hi! Would like to add a section on how vite /w ruby works internally under the hood and how it hooks in rails and the build process, are there any pointers on explaining what it _exactly does?_

Does it copy to the public folder? Or just pass requests trough? Etc. A couple of bullet points with the general idea would help a lot.

**Comment by pocheptsov** [comment]
@emilebosch that section would be definitely useful but under the hood `vite_ruby` works very similar to `webpack/webpacker` integration. @ElMassimo could obviously correct me.

**Comment by ElMassimo** [comment, marked as answer]
Hi @emilebosch, thanks for the suggestion!

As @pocheptsov mentioned, it's very similar to how webpacker works—specially from a user standpoint.

I've added an [Overview](https://vite-ruby.netlify.app/overview) section that provides a short reference to how it works. In particular, I differentiate which features are provided by Vite.js and which by Vite Ruby.

**Reply by emilebosch** [reply]
Perfect thank you very much for the addition!
